vscode/src/ruby/chruby.ts

```diff
@@ -156,7 +176,195 @@ export class Chruby extends VersionManager {
+        await new Promise<void>((resolve) => {
+          setTimeout(resolve, 10000);
+
+          // If the user cancels the fallback, resolve immediately so that they don't have to wait 10 seconds
+          token.onCancellationRequested(() => {
+            resolve();
+          });
+        });
+
+        if (token.isCancellationRequested) {
+          await this.handleCancelledFallback();
+
+          // We throw this error to be able to catch and re-run activation after the user has configured a fallback
+          throw new RubyVersionCancellationError();
+        }
+
+        const fallback = await this.findFallbackRuby();
+
+        if (!fallback) {
+          throw new Error("Cannot find any Ruby installations");
+        }
+
+        return fallback;
+      },
+    );
+
+    return fallback;
+  }
+
+  private async handleCancelledFallback() {
+    const answer = await vscode.window.showInformationMessage(
+      `The Ruby LSP requires a Ruby version to launch.
+      You can define a fallback for the system or for the Ruby LSP only`,
+      "System",
+      "Ruby LSP only",
+    );
+
+    if (answer === "System") {
+      await this.createParentRubyVersionFile();
+    } else if (answer === "Ruby LSP only") {
+      await this.manuallySelectRuby();
+    }
+
+    throw new Error(
+      `Cannot find .ruby-version file. Please specify the Ruby version in a
+           .ruby-version either in ${this.bundleUri.fsPath} or in a parent directory`,
+    );
+  }
+
+  private async createParentRubyVersionFile() {
+    const items: vscode.QuickPickItem[] = [];
+
+    for (const uri of this.rubyInstallationUris) {
+      let directories;
+
+      try {
+        directories = (await vscode.workspace.fs.readDirectory(uri)).sort(
+          (left, right) => right[0].localeCompare(left[0]),
+        );
+
+        directories.forEach((directory) => {
+          items.push({
+            label: directory[0],
+          });
+        });
+      } catch (error: any) {
+        continue;
+      }
+    }
+
+    const answer = await vscode.window.showQuickPick(items, {
+      title: "Select a Ruby version to use as fallback",
+      ignoreFocusOut: true,
+    });
+
+    if (!answer) {
+      throw new Error(
+        `Cannot find .ruby-version file. Please specify the Ruby version in a
+           .ruby-version either in ${this.bundleUri.fsPath} or in a parent directory`,
+      );
+    }
+
+    const targetDirectory = await vscode.window.showOpenDialog({
+      defaultUri: vscode.Uri.file(os.homedir()),
+      openLabel: "Add fallback in this directory",
+      canSelectFiles: false,
+      canSelectFolders: true,
+      canSelectMany: false,
+      title: "Select the directory to create the .ruby-version fallback in",
+    });
+
+    if (!targetDirectory) {
+      throw new Error(
+        `Cannot find .ruby-version file. Please specify the Ruby version in a
+           .ruby-version either in ${this.bundleUri.fsPath} or in a parent directory`,
```

Comment:
It looks like we can surface this exact message in multiple occasions, can it be generated through a function to avoid the duplication?